core/email.go

```diff
@@ -16,22 +16,26 @@ var (
 	ErrEmailAddressEmpty   = errors.New("e-mail address is empty")
 )
 
-type EmailAddress interface {
-	// String returns the string representation for this e-mail address
-	String() string
-}
-
-type emailAddress struct {
+type EmailAddress struct {
 	address string
 }
 
-func (email emailAddress) String() string {
+func (email *EmailAddress) String() string {
```

Comment:
Is er een reden waarom je deze methodes op de pointers definieert en niet op het object zelf? Dat wordt normaal gebruikt voor methodes die het object muteren, maar bij e-mailadressen willen we dat nooit denk ik. Dan is het duidelijker als dat aan het object zelf hangt ipv aan een pointer. Voor hoe je ze oproept zou dat geen verschil mogen maken denk ik

Reply:
Ik ben ook niet zeker of die `String()` methode zelfs überhaupt nog nodig is en of je niet ook gewoon `string(emailaddress)` kan doen. Die functie was er vooral omdat een interface een functie nodig heeft (tenzij je expliciet `any` wilt). Ze mag gerust ook blijven bestaan van mij, ik kan mij inbeelden dat er nogal veel refactoring nodig zou zijn om die te verwijderen